cpp/log/cert.cc

```diff
@@ -233,8 +262,8 @@ string Cert::PrintTime(ASN1_TIME* when) {
 }
 
 
-Cert::Status Cert::IsIdenticalTo(const Cert& other) const {
-  return X509_cmp(x509_, other.x509_) == 0 ? TRUE : FALSE;
+StatusOr<bool> Cert::IsIdenticalTo(const Cert& other) const {
+  return StatusOr<bool>(X509_cmp(x509_, other.x509_) == 0 ? true : false);
```

Comment:
I think it would be safer if I don't introduce StatusOr<int> returns. It
would be cleaner to use this as we could get rid of result pointer
parameters but if we don't use multiple integer return types with StatusOr
there's much less chance of accidentally returning the wrong thing. For
these it can just return util::Status.

On 28 August 2015 at 12:58, Pierre Phaneuf <notifications@github.com> wrote:

> In cpp/log/cert.cc
> <https://github.com/google/certificate-transparency/pull/885#discussion_r38192578>
> :
>
> > @@ -233,8 +262,8 @@ string Cert::PrintTime(ASN1_TIME* when) {
> >  }
> >
> >
> > -Cert::Status Cert::IsIdenticalTo(const Cert& other) const {
> > -  return X509_cmp(x509_, other.x509_) == 0 ? TRUE : FALSE;
> > +StatusOr<bool> Cert::IsIdenticalTo(const Cert& other) const {
> > +  return StatusOr<bool>(X509_cmp(x509_, other.x509_) == 0 ? true : false);
>
> You're not safe from conversion between integer types, unfortunately. The
> following all compile, in a function with a StatusOr<bool>:
>
> return true;return 42;return StatusOr<bool>(23);return StatusOr<int>(77);
>
> So there's no gain to be made?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/certificate-transparency/pull/885/files#r38192578>
> .
>